### ISSUE TITLE
Implement Redis or alternative Fast key/value caches For Drupal/Cantaloupe

### ISSUE THREAD

**Issue by DiegoPino** [esmero/archipelago-deployment-live#6]
# What?

We love caching. Caching is the difference between lame responses and near realtime ones. MYSQL based cache For Drupal works, but e.g dedicated systems like Redis are faster: https://redis.io

Redis is an in memory cache/structure/hash store and we can deploy as part of your containerized/pod-ized deployment to make large sites even faster.

Redis can be reused also as cache for cantaloupe so double usage and that makes it a perfect candidate.

This requires testing/setting up/and learning how to configure correctly based on the required specs. So really a meta issue. But not exploratory but more like: let's go for this @pcambra @giancarlobi

**Comment by giancarlobi** [comment]
@DiegoPino Don't sure if useful, I installed redis for nextcloud deployment as stated here https://docs.nextcloud.com/server/20/admin_manual/configuration_server/caching_configuration.html#id2. ONLY installed so I know it works but I don't know how it works...

**Comment by DiegoPino** [comment]
yes! The setup is simple enough? And For Drupal its basically composer + settings https://www.drupal.org/project/redis
But we may want to measure performance! https://en.wikipedia.org/wiki/ApacheBench will be our friend here
and we can compare with/without!

For IIIF https://cantaloupe-project.github.io/manual/4.1/caching.html#RedisCache

**Comment by DiegoPino** [comment]
I will create a docker-compose test environment here for that purpose using "redis:alpine3.13" and then report back. This also goes hand by hand with an experiment we want to make, an 1,000,000 ADOs test environment to make sure performance is at its best (and I will need some help!)

**Comment by DiegoPino** [comment]
Also, for ease of use I will go for https://github.com/predis/predis/releases/tag/v1.1.6 which does not require to compile an PHP extension. If this works fine we can evaluate that rebuilding esmero-php:7.4RC2

**Comment by pcambra** [comment]
We need to install PhpRedis extension on the PHP side which is the recommended option rather than Predis. Redis is a plug and play solution so it's a replacement but we need to establish which bins go to Redis and which don't. I'm sure we have more bins than a regular site but this is an example of what I have around:

```
if (extension_loaded('redis')) {
  $settings['redis.connection']['host'] = 'site.redis.' . $infra;
  $settings['redis.connection']['port'] = 22122;
  // Unique across infrastructures.
  $settings['cache_prefix'] = $dbname;
  $settings['redis.connection']['interface'] = 'PhpRedis';
  $settings['container_yamls'][] = DRUPAL_ROOT . 'sites/redis.services.yml';
  $settings['cache']['default'] = 'cache.backend.redis';
  # $settings['queue_default'] = 'queue.redis';
  // Always set the fast backend for bootstrap, discover and config, otherwise
  // this gets lost when redis is enabled.
  $settings['cache']['bins']['bootstrap'] = 'cache.backend.chainedfast';
  $settings['cache']['bins']['discovery'] = 'cache.backend.chainedfast';
  # $settings['cache']['bins']['config'] = 'cache.backend.chainedfast';
  $settings['cache']['bins']['config'] = 'cache.backend.redis';
}
```
